Sources/SwiftSyntax/Convenience.swift

```diff
@@ -68,6 +68,73 @@ extension EnumCaseParameterSyntax {
   }
 }
 
+extension FloatLiteralExprSyntax {
+  public var floatingValue: Double? {
+    let floatingDigitsWithoutUnderscores = literal.text.filter {
+      $0 != "_"
+    }
+
+    return Double(floatingDigitsWithoutUnderscores)
+  }
+}
+
+extension IntegerLiteralExprSyntax {
+  public enum Radix {
+    case binary
+    case octal
+    case decimal
+    case hex
+
+    public var size: Int {
+      switch self {
+      case .binary: return 2
+      case .octal: return 8
+      case .decimal: return 10
+      case .hex: return 16
+      }
+    }
+
+    fileprivate var offset: Int {
+      switch self {
+      case .binary:
+        return 2
+      case .octal:
+        return 2
+      case .hex:
+        return 2
+      case .decimal:
+        return 0
+      }
+    }
+  }
+
+  public var radix: Radix {
+    let text = self.literal.text
+    if text.starts(with: "0b") {
+      return .binary
+    } else if text.starts(with: "0o") {
+      return .octal
+    } else if text.starts(with: "0x") {
+      return .hex
+    } else {
+      return .decimal
+    }
+  }
+
+  public var integerValue: Int? {
```

Comment:
A doc comment would be good here, especially detailing why this can be `nil`.

Same for `floatingValue`.

Reply:
I've tried to add something, could you please verify if that was something you had in mind?